src/main/java/pl/ms/springsecurity/user/UserController.java

```diff
@@ -0,0 +1,30 @@
+package pl.ms.springsecurity.user;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+
+
+@Controller
+class UserController {
+    private final UserService userService;
+
+    UserController(UserService userService) {
+        this.userService = userService;
+    }
+
+    @GetMapping("/userPanel")
+    public String userEditForm(Model model) {
+        User user = userService.getCurrentUser();
+        model.addAttribute("user", user);
+        return "userEditForm";
+    }
+
+    @PostMapping("/userPanel/edit/{id}")
+    public String updateUserData(@PathVariable Long id, User user) {
```

Comment:
Jeżeli korzystasz z obiektów DTO, to powinnaś być już raczej konsekwentna i powinno być tak, że masz jasno wyznaczone warstwy Encje > Repozytoria > Service > Kontrolery. Tutaj jest raz tak, raz tak.
Użycie tutaj typu User jest ryzykowne, bo przy odpowiednio zmienionym formularzu edycji dajesz tutaj możliwość ustawienia roli, co w przypadku błędnie zaimplementowanej warstwy usług może sprawić spore problemy z bezpieczeństwem. Postaram się jutro dograć jeszcze filmik do tego fragmentu